### PR TITLE
Add dynamic totals to diary

### PR DESCRIPTION
**Pull request by TheKaterTot**
This adds the totals table to the diary page, and adds a totals row to each meal table. They are dynamic, and they adjust with the day.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -36,19 +36,39 @@ function createDiaryURL(date, category) {
...
+    createFoodHandler('#dinner')(data)
+    totalMealCalories('#dinner');
+    dailyCalories();
+    remainingDailyCalories();
+  })
 }
 
 function fetchSnacks(date) {
-  $.getJSON(createDiaryURL(date, 'snacks'), createFoodHandler('#snacks'));
+  $.getJSON(createDiaryURL(date, 'snacks'), (data) => {
```
Comment (cmacaulay) [inline lib/index.js:66]: These functions aren't feeling super DRY, do you think we could refactor into a method that passes the meal name through as an argument?

File: lib/index.js [hunk 2]
```diff
@@ -70,3 +90,28 @@ function nextDay() {
...
+    total += parseInt($(calories).text());
+  })
+  $('.calories-consumed').text(total);
+  return total;
+}
+
+function remainingDailyCalories() {
+  let goal = parseInt($('.goal-calories').text());
+  let total = parseInt(dailyCalories());
+  $('.calories-remaining').text(goal - total);
```
Comment (cmacaulay) [inline lib/index.js:116]: 🎉 🎉 math!

File: main.css [hunk 3]
```diff
@@ -1038,9 +1038,14 @@ h1 {
...
+<<<<<<< HEAD
 .error {
   color: #af125a;
   font-size: 12px;
   padding-bottom: 5px;
   text-align: center;
+=======
+.bold-text {
+  font-weight: bold;
+>>>>>>> Add dynamic totals to diary
```
Comment (cmacaulay) [inline main.css]: @TheKaterTot this looks like a merge conflict?

**Review by cmacaulay — CHANGES_REQUESTED** [review]
@TheKaterTot looks good! Saw one area for potential refactoring, and it looks like there might be a merge conflict still in main.css(which is weird because that file should be in the .gitignore?)

File: lib/index.js (hunk 1 above)
Comment (TheKaterTot) [reply lib/index.js:66]: Yeah, they are still a bit repetitive, but the async stuff was tricky here, and there were only so many ways I could rearrange it where it happened in the necessary order. I'll keep it on my list of refactors, but I'm going to move forward for now.

**Comment by TheKaterTot** [comment]
The style docs are acting a little weird in the .gitignore and in the merge conflict. Maybe something to do with the compiling?

**Pull request merged**